### PR TITLE
RoR Group Project

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS MILESTONE WE HAVE SUCCESSFULLY:
1. Added a login page.
2. Added registration page.
3. Added food list (with all CRUD implementation, except for 'update').
4. Added recipes list (with all CRUD implementation, except for 'update').
5. Added general shopping list view (all missing food for all our recipes and total price).
6. Added list of all public recipes from other users with their names and total prices.
7. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
8. Made sure that we used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
9. Made sure that we documented our work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
10. Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @Kgomotso196 and @akezeth, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Pixel Perfect Design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  has_many :recipes, dependent: :destroy, class_name: 'Recipe'
+  has_many :recipe_foods, dependent: :destroy
+  has_many :foods, dependent: :destroy
+
+  def admin?
+    role == 'admin'
+  end
```
[the remark this answers is not in the document]
Comment (Kgomotso196) [reply app/models/user.rb:12]: Hello @Strangeal , I appreciate your review. However, I'm not sure what error you are encountering on your side but the role column already exists for the user table. Here's a screenshot for reference.
users

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    <div class="recipe_box_left">
+      <p>Preparation time: <%= @recipe.preparation_time %> min</p>
+      <d>Cooking time: <%= @recipe.cooking_time %> min</p>
+      <p>Steps go here...</p>
+    </div>
+    <div class="recipe_box_right">
+      <% if can? :update, @recipe %>
+        <%= button_to(@recipe.public? ? "Make Private" : "Make Public", toggle_recipe_public_path(@recipe), method: :put, class: "btn btn-info") %>
+      <% end %>
+    </div>
```
[the remark this answers is not in the document]
Comment (Kgomotso196) [reply app/views/recipes/show.html.erb]: @Strangeal Thank you for a constructive feedback, working on it.

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**